client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerInterfaceParser.java

```diff
@@ -30,14 +30,17 @@
     public SwaggerInterfaceParser(Class<?> swaggerInterface, String host) {
         this.swaggerInterface = swaggerInterface;
 
-        final Host hostAnnotation = swaggerInterface.getAnnotation(Host.class);
-        if (hostAnnotation == null) {
-            throw new MissingRequiredAnnotationException(Host.class, swaggerInterface);
-        }
-        else if (host != null) {
+        if (host != null && !host.isEmpty()) {
             this.host = host;
-        } else {
-            this.host = hostAnnotation.value();
+        }
+        else {
+            final Host hostAnnotation = swaggerInterface.getAnnotation(Host.class);
+            if (hostAnnotation != null && !hostAnnotation.value().isEmpty()) {
```

Comment:
any edge case where `hostAnnotation.value()` may be null hence expression `hostAnnotation.value().isEmpty()` throws NPE?

Reply:
If someone decorates their interface with this annotation and passes null to it... well, I just don't think they're doing themselves any favors.